src/packages/helpers/src/provider.ts

```diff
@@ -0,0 +1,126 @@
+import { Filter, PaginationOptions, BackendProvider } from '@exogee/graphweaver';
+
+export interface ProviderOptions<Entity, Context> {
+	init?(): Promise<Context>;
+	create?(context: Context, entity: Partial<Entity>): Promise<Entity>;
+	read(
+		context: Context,
+		filter: Filter<Entity>,
+		pagination?: PaginationOptions
+	): Promise<Entity | Array<Entity> | null>;
+	update?(context: Context, id: string, entity: Partial<Entity>): Promise<Entity>;
+	remove?(context: Context, filter: Filter<Entity>): Promise<boolean>;
+	backendId: string;
+}
+
+export const createProvider = <Entity, Context>(options: ProviderOptions<Entity, Context>) => {
+	class Provider<Entity, Context> implements BackendProvider<Entity, Entity> {
```

Comment:
I dont quite get the generics here `BackendProvider<Entity, Entity>` is expecting two types a GraphQL entity and a data entity. do we need a third generic here?

Reply:
i guess in this simple case we haven't made a distinction, the types are the same between the two- what i'll do is allow passing in of a third type which is the "data" backend type

Reply:
I've split these out but defaulted DataEntity to Entity for now